relay/relay.go

```diff
@@ -56,6 +56,8 @@ type CallFrame interface {
 	// of TChannel-Thrift Arg Scheme. If no iterator is available, return
 	// io.EOF.
 	Arg2Iterator() (arg2.KeyValIterator, error)
+	// Arg2Append appends a key/val pair to arg2
+	Arg2Append(key, val []byte) error
```

Comment:
As we discussed offline, this API may be limiting if we ever want to add more than one header(without the performance impact of moving arg3 around for each key/value we had), or for other types of modifications we want.

it also may not work well with the spill-over of arg3 into a new frame